### ISSUE TITLE
Path-aware ID matching

### ISSUE THREAD

**Issue by skeggse** [frostney/rollup-plugin-alias#47]
I'm seeing that an `alias` to `moment` will grab `moment-timezone` imports, despite having a later definition that specifies a different alias for `moment-timezone`:

```js
// In rollup.config.js
alias({
  moment: `${__dirname}/path/to/moment.js`,
  'moment-timezone': `${__dirname}/path/to/moment-timezone.js`,
});
```

```js
// Entry file.
import 'moment-timezone';
```

```sh
$ npx rollup -c rollup.config.js
Error: Could not load .../path/to/moment.js-timezone.js (imported by .../path/to/entry.js): ENOENT: no such file or directory, open '.../path/to/moment.js-timezone.js'
    at .../node_modules/rollup/dist/rollup.js:9976:10
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

In this example, reordering the two definitions passed to `alias` fixes the problem, though relying object property iteration order seems quite hacky. I'm not sure what the perfect solution here is, though I might expect the identifier provided to match the first path component, instead of a strict prefix.

**Comment by TriMoon** [comment]
@skeggse,
> In this example, reordering the two definitions passed to `alias` fixes the problem, though relying object property iteration order seems quite hacky. I'm not sure what the perfect solution here is, though I might expect the identifier provided to match the first path component, instead of a strict prefix.

The functionality seems to be "first match" as can be seen [here](https://github.com/rollup/rollup-plugin-alias/blob/841cd79a6e286e3bc1ddb75f9836395ab07c2032/src/index.js#L59-L60) and [logic](https://github.com/rollup/rollup-plugin-alias/blob/841cd79a6e286e3bc1ddb75f9836395ab07c2032/src/index.js#L12-L22) in the code :wink: 

It would be nice to be able to use [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)'s though...

**Comment by thiscantbeserious** [comment]
Update: This should've been solved by Version 2.0. 

Which moved to an array of objects for the definition and will keep the specific order of entries in addition to providing subpattern matching and replacement option via Regular Expressions 

(more on that see the Readme).

**Comment by shellscape** [comment]
Hey folks. This is a saved-form message, but rest assured we mean every word. The Rollup team is attempting to clean up the Issues backlog in the hopes that the active and still-needed, still-relevant issues bubble up to the surface. With that, we're closing issues that have been open for an eon or two, and have gone stale like pirate hard-tack without activity.

We really appreciate the folks have taken the time to open and comment on this issue. Please don't confuse this closure with us not caring or dismissing your issue, feature request, discussion, or report. The issue will still be here, just in a closed state. If the issue pertains to a bug, please re-test for the bug on the latest version of Rollup and if present, please tag @shellscape and request a re-open, and we'll be happy to oblige.
